Kitodo/src/main/java/de/sub/goobi/helper/tasks/ExportNewspaperBatchTask.java

```diff
@@ -350,20 +353,19 @@ private static String getMetsYearAnchorPointerURL(Process process)
      *
      * @return a list with the dates of all issues in this process
      */
-    private static List<LocalDate> getIssueDates(DigitalDocument act) {
+    private static List<LocalDate> getIssueDates(DigitalDocumentInterface act) {
         List<LocalDate> result = new LinkedList<>();
-        DocStruct logicalDocStruct = act.getLogicalDocStruct();
-        for (DocStruct annualNode : skipIfNull(logicalDocStruct.getAllChildren())) {
-            int year = getMetadataIntValueByName(annualNode, MetsModsImportExport.CREATE_LABEL_ATTRIBUTE_TYPE);
-            for (DocStruct monthNode : skipIfNull(annualNode.getAllChildren())) {
+        DocStructInterface logicalDocStruct = act.getLogicalDocStruct();
+        for (DocStructInterface annualNode : skipIfNull(logicalDocStruct.getAllChildren())) {
+            int year = getMetadataIntValueByName(annualNode, MetsModsImportExportInterface.CREATE_LABEL_ATTRIBUTE_TYPE);
+            for (DocStructInterface monthNode : skipIfNull(annualNode.getAllChildren())) {
                 int monthOfYear = getMetadataIntValueByName(monthNode,
-                        MetsModsImportExport.CREATE_ORDERLABEL_ATTRIBUTE_TYPE);
-                for (DocStruct dayNode : skipIfNull(monthNode.getAllChildren())) {
-                    LocalDate appeared = new LocalDate(year, monthOfYear,
-                            getMetadataIntValueByName(dayNode, MetsModsImportExport.CREATE_ORDERLABEL_ATTRIBUTE_TYPE));
-                    for (
-                        @SuppressWarnings("unused")
-                        DocStruct entry : skipIfNull(dayNode.getAllChildren())) {
+                    MetsModsImportExportInterface.CREATE_ORDERLABEL_ATTRIBUTE_TYPE);
+                for (DocStructInterface dayNode : skipIfNull(monthNode.getAllChildren())) {
+                    LocalDate appeared = new LocalDate(year, monthOfYear, getMetadataIntValueByName(dayNode,
+                        MetsModsImportExportInterface.CREATE_ORDERLABEL_ATTRIBUTE_TYPE));
+                    for (@SuppressWarnings("unused")
```

Comment:
Why actually this SuppressWarnings is here?

Reply:
I believe to remember that we have discussed about this one before. It is there because the variable is not used. The program relies only on the correct number of iterations.

Reply:
Yes, I remind myself. 

So correctly it should look like that:
for (@SuppressWarnings("unused") DocStructInterface entry : skipIfNull(dayNode.getAllChildren())) {
                        result.add(appeared);
                    }